src/driver/franka_plan_runner.cc

```diff
@@ -738,6 +760,67 @@ void FrankaPlanRunner::IncreaseFrankaTimeBasedOnStatus(
   }
 }
 
+bool FrankaPlanRunner::IsStartFarFromCurrentJointPosition(
+    const RobotParameters& params, const Eigen::VectorXd& franka_start_conf,
+    const Eigen::VectorXd& start_conf_plan) {
+  // Maximum change in joint angle between two confs
+  auto max_ang_distance {
+      utils::max_angular_distance(franka_start_conf, start_conf_plan)};
+  if (max_ang_distance > params.kMediumJointDistance) {
+    // far from start conf. return true
+    dexai::log()->error(
+        "JointPositionCallback: Discarding plan, mismatched start position."
+        " Max distance: {} > {}",
+        max_ang_distance, params.kMediumJointDistance);
+    return true;
+  }
+  // always returns false below this
+
+  // check to print warning. no control logic
+  if (max_ang_distance > params.kTightJointDistance) {
+    dexai::log()->warn(
+        "JointPositionCallback: max angular distance between franka and "
+        "start of plan is larger than 'kTightJointDistance': {} > {}",
+        max_ang_distance, params.kTightJointDistance);
+  }
+  // not far from start conf. return false
+  return false;
+}
+
+void FrankaPlanRunner::UpdateActivePlan(
+    std::unique_ptr<PPType> new_plan, int64_t new_plan_utime,
+    int64_t new_plan_exec_opt,
+    const Eigen::Vector3d& new_plan_contact_expected) {
+  const bool has_active_plan {plan_ != nullptr};
+  plan_ = std::move(new_plan);
+  plan_utime_ = new_plan_utime;
+  plan_exec_opt_ = new_plan_exec_opt;
+  contact_expected_ = new_plan_contact_expected;
+
+  plan_start_utime_ = utils::get_current_utime();
+
+  dexai::log()->info(
+      "JointPositionCallback: popped new plan {} from buffer, "
+      "starting initial timestep...",
+      plan_utime_);
+
+  if (!has_active_plan) {
```

Comment:
Shouldn't this have a trailing underscore?

Reply:
No, this is not a class member